source/entities/Creature.h

```diff
@@ -423,21 +428,18 @@ class Creature: public MovableGameEntity
     std::deque<CreatureAction>      mActionQueue;
     std::vector<Tile*>              mVisualDebugEntityTiles;
 
-
-    //! \brief Allows to keep track of the entity we are currently attacking
-    Tile*                           mAttackedTile;
-    GameEntity*                     mAttackedObject;
+    //! \brief Contains the actions that have already been tested to avoid trying several times same action
+    std::vector<CreatureAction::ActionType> mActionTry;
 
     ForceAction                     mForceAction;
 
-    bool                            mIsCarryActionTested;
     MovableGameEntity*              mCarriedEntity;
     GameEntity::ObjectType          mCarriedEntityDestType;
     std::string                     mCarriedEntityDestName;
 
     Ogre::Vector3                   mScale;
 
-    void pushAction(CreatureAction action);
+    bool pushAction(CreatureAction action, bool forcePush = false);
```

Comment:
Here I would document that the forcePush adds an action even if the action has been tried by the creature before.